subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestOptionsIntegrationSpec.groovy

```diff
@@ -0,0 +1,360 @@
+        |           useJUnitJupiter()
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       options {
+        |                           ${type} 'fast'
+        |                       }
+        |                   }
+        |               }
+        |           }
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        expect:
+        succeeds ":$task"
+
+        where:
+        suiteName   | suiteDeclaration              | task        | type
+        'test'      | 'test'                        | 'test'      | 'includeTags'
+        'test'      | 'test'                        | 'test'      | 'excludeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'includeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'excludeTags'
+    }
+
+    // See JUnitCategoriesIntegrationSpec for the inspiration for this test
+    def "re-executes test when #type is changed in #suiteName"() {
+        given:
+        resources.maybeCopy("TestOptionsIntegrationSpec")
+        buildFile << """
+        |testing {
+        |   suites {
+        |       $suiteDeclaration {
+        |           useJUnitJupiter()
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       options {
+        |                           ${type} 'fast'
+        |                       }
+        |                   }
+        |               }
+        |           }
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        when:
+        succeeds ":$task"
+
+        then:
+        executedAndNotSkipped ":$task"
+
+        when:
+        resources.maybeCopy("TestOptionsIntegrationSpec")
+        buildFile << """
+        |testing {
+        |   suites {
+        |       $suiteDeclaration {
+        |           useJUnitJupiter()
+        |           targets {
+        |               all {
+        |                   testTask.configure {
+        |                       options {
+        |                           ${type} 'slow'
+        |                       }
+        |                   }
+        |               }
+        |           }
+        |       }
+        |   }
+        |}""".stripMargin()
+
+        and:
+        succeeds ":$task"
+
+        then:
+        executedAndNotSkipped ":$task"
+
+        where:
+        suiteName   | suiteDeclaration              | task        | type
+        'test'      | 'test'                        | 'test'      | 'includeTags'
+        'test'      | 'test'                        | 'test'      | 'excludeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'includeTags'
+        'integTest' | 'integTest(JvmTestSuite)'     | 'integTest' | 'excludeTags'
+    }
+
+    // Migrated from TestSuitesIntegrationTest to consolidate options-related tests
```

Comment:
Not sure this comment is necessary within the actual source. IMO we should update this test to be consistent with the rest of the tests in the class, now that there is a more formal pattern.

Reply:
I'll update this.